psi4/src/psi4/adc/adc.cc

```diff
@@ -45,53 +45,48 @@ ADCWfn::ADCWfn(SharedWavefunction ref_wfn, Options& options) : Wavefunction(opti
+    aoccpi_ = nalphapi_ - frzcpi_;
+    boccpi_ = nbetapi_ - frzcpi_;
+    avirpi_ = nmopi_ - frzvpi_ - nalphapi_;
+    bvirpi_ = nmopi_ - frzvpi_ - nbetapi_;
+
+    int naocc = aoccpi_.sum();
+    int nbocc = boccpi_.sum();
+    int navir = avirpi_.sum();
+    int nbvir = bvirpi_.sum();
 
-    int naocc = 0, nbocc = 0, navir = 0, nbvir = 0;
     int aoccount = 0, boccount = 0, avircount = 0, bvircount = 0;
-    for (int h = 0; h < nirrep_; h++) {
-        aoccpi_[h] = doccpi_[h] + soccpi_[h] - frzcpi_[h];
-        boccpi_[h] = doccpi_[h] - frzcpi_[h];
-        avirpi_[h] = nmopi_[h] - doccpi_[h] - soccpi_[h] - frzvpi_[h];
-        bvirpi_[h] = nmopi_[h] - doccpi_[h] - frzvpi_[h];
-
-        naocc += aoccpi_[h];
-        nbocc += boccpi_[h];
-        navir += avirpi_[h];
-        nbvir += bvirpi_[h];
-    }
+
     aocce_ = new double[naocc];
     bocce_ = new double[nbocc];
     avire_ = new double[navir];
     bvire_ = new double[nbvir];
 
-    nopen_ = 0;
-    for (int h = 0; h < nirrep_; h++) nopen_ += soccpi_[h];
+    nopen_ = soccpi().sum();
     if (!psio_) {
         throw PSIEXCEPTION("The wavefunction passed in lacks a PSIO object, crashing ADC. See GitHub issue #1851.");
     }
-    if (nopen_) throw PSIEXCEPTION("Openshell calculation has not been implemented yet!");
+    if (nopen_) throw PSIEXCEPTION("Open-shell calculation has not been implemented yet!");
 
     aoccount = 0, boccount = 0, avircount = 0, bvircount = 0;
     for (int h = 0; h < nirrep_; h++) {
-        for (int a = frzcpi_[h]; a < doccpi_[h] + soccpi_[h]; a++) aocce_[aoccount++] = epsilon_a_->get(h, a);
+        for (int a = frzcpi_[h]; a < nalphapi_[h]; a++) aocce_[aoccount++] = epsilon_a_->get(h, a);
 
-        for (int b = frzcpi_[h]; b < doccpi_[h]; b++) bocce_[boccount++] = epsilon_b_->get(h, b);
+        for (int b = frzcpi_[h]; b < nbetapi_[h]; b++) bocce_[boccount++] = epsilon_b_->get(h, b);
 
-        for (int a = doccpi_[h] + soccpi_[h]; a < nmopi_[h] - frzvpi_[h]; a++)
+        for (int a = nalphapi_[h]; a < nmopi_[h] - frzvpi_[h]; a++)
             avire_[avircount++] = epsilon_a_->get(h, a);
 
-        for (int b = doccpi_[h]; b < nmopi_[h] - frzvpi_[h]; b++) bvire_[bvircount++] = epsilon_b_->get(h, b);
+        for (int b = nbetapi_[h]; b < nmopi_[h] - frzvpi_[h]; b++) bvire_[bvircount++] = epsilon_b_->get(h, b);
     }
 
+    const auto docc = doccpi();
+    const auto socc = soccpi();
     outfile->Printf("\n\n\tIrrep  Core  Docc  Socc  aOcc  aVir  bOcc  bVir  FVir\n");
     outfile->Printf("\t*****************************************************\n");
     for (int h = 0; h < nirrep_; h++) {
         outfile->Printf("\t %3s   %3d   %3d   %3d   %3d   %3d   %3d   %3d   %3d\n", irreps_[h].c_str(), frzcpi_[h],
-                        doccpi_[h], soccpi_[h], aoccpi_[h], avirpi_[h], boccpi_[h], bvirpi_[h], frzvpi_[h]);
+                        docc[h] + socc[h], soccpi()[h], aoccpi_[h], avirpi_[h], boccpi_[h], bvirpi_[h], frzvpi_[h]);
```

Comment:
You're supposed to print docc and socc so
```suggestion
                        docc[h], socc[h], aoccpi_[h], avirpi_[h], boccpi_[h], bvirpi_[h], frzvpi_[h]);
```